domain/config.go

```diff
@@ -121,6 +245,48 @@ func viperDecodeHookFunc() mapstructure.DecodeHookFunc {
 	}
 }
 
+// Automatically generate environment variable mappings for all fields,
+// including nested structs, without having to manually specify each binding.
+// For example, if a mapstructure tag is "foo", the environment variable will be "SQS_FOO".
+// If nested structs are present such as "foo.bar", the environment variable will be "SQS_FOO_BAR".
+func bindEnvRecursive(v reflect.Value, prefix string) {
+	t := v.Type()
+
+	// Assume pointer to struct
+	if t.Kind() == reflect.Ptr {
+		v = v.Elem()
+		t = v.Type()
+	}
+
+	// If not a struct after dereferencing, return
+	if t.Kind() != reflect.Struct {
+		return
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		value := v.Field(i)
+
+		// Get the mapstructure tag, if any
+		tag := field.Tag.Get("mapstructure")
+		if tag == "" {
+			tag = strings.ToLower(field.Name)
+		}
+
+		envName := prefix + tag
+
+		// For nested structs, recurse
+		if value.Kind() == reflect.Struct || (value.Kind() == reflect.Ptr && value.Elem().Kind() == reflect.Struct) {
+			bindEnvRecursive(value, envName+".")
+		} else {
+			// Bind the environment variable
+			if err := viper.BindEnv(envName); err != nil {
+				panic(err)
+			}
+		}
+	}
+}
+
```

Comment:
Note to reviewer: this logic handles dynamically creating environment variable bindings to to allow overriding any config value from the default config